test/tests/session.ts

```diff
@@ -31,95 +37,87 @@ suite('session', function () {
-                const testSession = new Session({
-                    ...mockSessionOptions,
-                    hooks: {
-                        afterSign: [new MockTransactHook()],
-                    },
-                })
-                assert.lengthOf(testSession.hooks.afterBroadcast, 0)
-                assert.lengthOf(testSession.hooks.afterSign, 1)
-                assert.lengthOf(testSession.hooks.beforeBroadcast, 0)
-                assert.lengthOf(testSession.hooks.beforeSign, 0)
-            })
-            test('assign beforeBroadcast', function () {
-                const testSession = new Session({
-                    ...mockSessionOptions,
-                    hooks: {
-                        beforeBroadcast: [new MockTransactHook()],
-                    },
+                test('inherit', async function () {
+                    const sessionKit = new SessionKit({
+                        appName: 'demo.app',
+                        chains: [
+                            {
+                                id: '2a02a0053e5a8cf73a56ba0fda11e4d92e0238a4a2aa74fccf46d5a910746840',
+                                url: 'https://jungle3.greymass.com',
+                            },
+                        ],
+                        fetch,
+                        transactPlugins: [new MockTransactPlugin()],
+                        walletPlugins: [makeWallet()],
+                    })
+                    const session = await sessionKit.login()
+                    assert.instanceOf(session, Session)
+                    assert.lengthOf(session.transactPlugins, 1)
+                    assert.instanceOf(session.transactPlugins[0], MockTransactPlugin)
                 })
-                assert.lengthOf(testSession.hooks.afterBroadcast, 0)
-                assert.lengthOf(testSession.hooks.afterSign, 0)
-                assert.lengthOf(testSession.hooks.beforeBroadcast, 1)
-                assert.lengthOf(testSession.hooks.beforeSign, 0)
-            })
-            test('assign beforeSign', function () {
-                const testSession = new Session({
-                    ...mockSessionOptions,
-                    hooks: {
-                        beforeSign: [new MockTransactHook()],
-                    },
+                test('override', async function () {
+                    const sessionKit = new SessionKit({
+                        appName: 'demo.app',
+                        chains: [
+                            {
+                                id: '2a02a0053e5a8cf73a56ba0fda11e4d92e0238a4a2aa74fccf46d5a910746840',
+                                url: 'https://jungle3.greymass.com',
+                            },
+                        ],
+                        fetch,
+                        walletPlugins: [makeWallet()],
+                    })
+                    const session = await sessionKit.login({
+                        transactPlugins: [new MockTransactPlugin()],
```

Comment:
The `TransactPlugins` can be overridden during a specific `.login` call if desired.